src/rpc.ts

```diff
@@ -45,16 +48,54 @@ router.post('/', async (req, res) => {
       return rpcError(res, 500, 'Unable to authenticate your verification link', id);
     } else if (method === 'snapshot.unsubscribe') {
       if (verifyUnsubscribe(params.email, params.signature)) {
-        await unsubscribe(params.email);
+        await unsubscribe(params.email, params.subscriptions);
         return rpcSuccess(res, 'OK', id);
       }
 
       return rpcError(res, 500, 'Unable to authenticate your verification link', id);
+    } else if (method === 'snapshot.update') {
+      if (!Array.isArray(params.subscriptions)) {
+        return rpcError(res, 400, 'Invalid params', id);
+      }
+
+      if (verifyUpdate(params.email, params.address, params.subscriptions, params.signature)) {
+        await update(params.email, params.address, params.subscriptions);
+        return rpcSuccess(res, 'OK', id);
+      }
+
+      return rpcError(res, 500, 'Unable to authenticate your request', id);
     }
   } catch (e) {
     console.log(e);
     return rpcError(res, 500, e, id);
   }
 });
 
+router.post('/subscriber', async (req, res) => {
+  const { address } = req.body;
+
+  try {
+    const subscribers = await getAddressSubscriptions(address);
+    if (!subscribers || !subscribers[0]) {
```

Comment:
1 wallet can be attached to only on email

Reply:
than something wrong, because I have succeeded to register with multiple emails:
 
<img width="491" alt="Screenshot 2023-05-12 at 19 58 25" src="https://github.com/snapshot-labs/envelop/assets/7693369/2c5efa7e-3d70-4a6b-b324-1ad0c2b60bfa">

Reply:
This is expecting that the PR https://github.com/snapshot-labs/envelop/pull/52 was pulled, which add a unique index on these columns preventing duplicate.

By the way, without the new unique index, isn't the primary key supposed to prevent that too ?

Reply:
yes, but in PR #52 you have added a `UNIQUE` index, but this will not fix the case. Because you still will be able to attach different emails to the same address. I have added the comment to #52 
If you want to prevent this, you need to add `UNIQUE` index to the address column only.

Reply:
You are correct, I'll add a check

Reply:
> what if the client subscribed to multiple emails? Maybe we should handle it on the step of subscription and notify him/her that he already subscribed with another email?

He can subscribe to many emails, but he can only verify only one. All subsequent verification should fail, since one address can only be associated to a verified email.

Reply:
Checked. Looks like it is possible. I verified multiple emails:

<img width="847" alt="Screenshot 2023-05-12 at 21 28 28" src="https://github.com/snapshot-labs/envelop/assets/7693369/3f81a62d-5333-4f65-9e9f-a8f0e65a7e84">

btw, I don't use UI, I call all endpoints with curl

Reply:
WIP, will prevent verifying multiple emails on the same address

Reply:
Done, only a single email can be verified per wallet

Reply:
Additionally, you can also check that from UI side, by querying /subscriber with the wallet address